streamer/src/nonblocking/quic.rs

```diff
@@ -153,7 +153,8 @@ async fn setup_connection(
                 }
                 connection.set_max_concurrent_uni_streams(
                     VarInt::from_u64(
-                        ((stake as f64 / total_stake as f64) * QUIC_TOTAL_STAKED_CONCURRENT_STREAMS)
+                        ((stake as f64 / total_stake as f64)
+                            * QUIC_TOTAL_STAKED_CONCURRENT_STREAMS as f64)
```

Comment:
pretty sure we can get away without the casts

```suggestion
                        stake
                            .saturating_mul(QUIC_TOTAL_STAKED_CONCURRENT_STREAMS
                            .checked_div(total_stake)
                            .unwrap_or(0)
```

Reply:
+1 on this approach, but we'll have to widen the types to `u128` during multiplication.  For example, if we have 1 billion SOL, this would overflow. Or at least, the compiler told me it would.

Reply:
oh actually thought we had an extra couple orders of stake that a validator would have to obtain to start saturating, but we're there today.  i am bad at math.

so yeah need to go to `u128` here. I'll updated the suggestion

Reply:
I think we do not want to stake * QUIC_TOTAL_STAKED_CONCURRENT_STREAMS so easily to saturate